File: WebApplication/src/modules/student/dtos/person-lot.dtos/person-basic.dtos/person-basic.dto.ts
import { AttachedFile } from "src/infrastructure/models/attached-file.model";
import { NomenclatureCodeDto } from "src/infrastructure/models/nomenclature-code.dto";
import { NomenclatureDto } from "src/infrastructure/models/nomenclature.dto";
import { GenderType } from "src/modules/student/enums/gender-type.enum";

export class PersonBasicDto {
  firstName: string;
  middleName: string;
  lastName: string;
  otherNames: string;

  firstNameAlt: string;
  middleNameAlt: string;
  lastNameAlt: string;
  otherNamesAlt: string;

  uin: string;
  foreignerNumber: string;
  idnNumber: string;

  email: string;
  phoneNumber: string;

  gender: GenderType;
  birthDate: Date;

  birthCountry: NomenclatureCodeDto;
  birthDistrict: NomenclatureCodeDto;
  birthMunicipality: NomenclatureCodeDto;
  birthSettlement: NomenclatureCodeDto;

  citizenship: NomenclatureDto;
  secondCitizenship: NomenclatureDto;

  citizenshipFull: string;
  citizenshipFullAlt: string;

  residenceCountry: NomenclatureCodeDto;
  residenceDistrict: NomenclatureCodeDto;
  residenceMunicipality: NomenclatureCodeDto;
  residenceSettlement: NomenclatureCodeDto;
  residenceAddress: string;
  postCode: string;

  personImage: AttachedFile;

  residenceSettlementAndPostCode: string;
  residenceSettlementAndPostCodeAlt: string;
}

File: StudentCard.Application/Students/Dtos/PersonLotDtos/PersonBasicDtos/PersonBasicDto.cs
using StudentCard.Data.Rdpzsd.Enums;
using StudentCard.Data.Rdpzsd.Students.Parts;
using StudentCard.Infrastructure.Helpers.Dtos;
using System;

namespace StudentCard.Application.Students.Dtos.PersonLotDtos.PersonBasicDtos
{
    public class PersonBasicDto
    {
        public string FirstName { get; set; }
        public string MiddleName { get; set; }
        public string LastName { get; set; }
        public string OtherNames { get; set; }

        public string FirstNameAlt { get; set; }
        public string MiddleNameAlt { get; set; }
        public string LastNameAlt { get; set; }
        public string OtherNamesAlt { get; set; }

        public string Uin { get; set; }
        public string ForeignerNumber { get; set; }
        public string IdnNumber { get; set; }

        public string Email { get; set; }
        public string PhoneNumber { get; set; }
        public string PostCode { get; set; }

        public GenderType Gender { get; set; }
        public DateTime BirthDate { get; set; }

        public NomenclatureCodeDto BirthCountry { get; set; }
        public NomenclatureCodeDto BirthDistrict { get; set; }
        public NomenclatureCodeDto BirthMunicipality { get; set; }
        public NomenclatureCodeDto BirthSettlement { get; set; }
        public string ForeignerBirthSettlement { get; set; }

        public NomenclatureDto Citizenship { get; set; }
        public NomenclatureDto SecondCitizenship { get; set; }

        public string CitizenshipFull { get; set; }
        public string CitizenshipFullAlt { get; set; }

        public NomenclatureCodeDto ResidenceCountry { get; set; }
        public NomenclatureCodeDto ResidenceDistrict { get; set; }
        public NomenclatureCodeDto ResidenceMunicipality { get; set; }
        public NomenclatureCodeDto ResidenceSettlement { get; set; }
        public string ResidenceAddress { get; set; }

        public string ResidenceSettlementAndPostCode { get; set; }
        public string ResidenceSettlementAndPostCodeAlt { get; set; }

        public PersonImage PersonImage { get; set; }
    }
}
